components/NavigationBar.vue

```diff
@@ -48,9 +51,17 @@ export default {
   components: {
     SideMenu
   },
+  mounted() {
+    this.changeRoutingIfAdmin()
+  },
   methods: {
     setViewMenu() {
       this.viewMenu = !this.viewMenu;
+    },
+    async changeRoutingIfAdmin() {
```

Comment:
Não entendi esse método, mas se for pra fazer redirecionamento de módulos, isso já existe no middleware principal auth.guard. Quanto a recuperação do token pra setar nas requisições você pode usar o generics, lá ja tem um interceptor fazendo isso.